tests/test_outputs/test_intram_by_pg.py

```diff
@@ -1,178 +1,210 @@
 """Tests for intram_by_pg.py."""
 # Standard Library Imports
-import pytest
-import os
-import logging
-import pathlib
 
-# Third Party Imports
+from pandas.testing import assert_frame_equal
+import logging
 import pandas as pd
+import unittest
 
 # Local Imports
 from src.outputs.intram_by_pg import output_intram_by_pg
-from tests.test_outputs.conftest import read_config
-
-
-# Assign config values to paths
-config = read_config()
-LOCATION = config["global"]["platform"]
+from src.outputs.intram_by_pg import _generate_intarm_by_pg
 
 # create logger (required pass to function)
 TestLogger = logging.getLogger(__name__)
 
 
-class TestOutputIntramByPG(object):
-    """Tests for output_intram_by_pg."""
+class TestIngramBYPg(unittest.TestCase):
+    """Unit tests for:
+    - intram_by_pg._output_intram_by_pg
+    - intram_by_pg.generate_intarm_by_pg.
+    """
+    def create_config_dict(self):
+        """Creates config dictionary for test"""
+        config = {"outputs_paths": {"outputs_master": "test_path"},
+                  "years": {"survey_year": "2023"}}
+        return config
+    
+    def mock_write_csv(self, filepath: str, data: pd.DataFrame) -> None:
+        """Dummy script mimicking `s3_mods.rc_write_csv`.
+
+        Args:
+            filepath (str): The filepath to save the DataFrame to.
+            data (pd.DataFrame): The DataFrame to write to the passed path.
 
-    def setup_tmp_dir(self, path: pathlib.Path, ni: bool = True) -> pathlib.Path:
-        """Set up the output directory given a temp path."""
-        # create output dirs
-        output_parent = os.path.join(path, "10_outputs")
-        output_child = os.path.join(
-            output_parent, f"output_intram_by_pg_{'gb' if not ni else 'uk'}"
-        )
-        os.makedirs(output_child)
-        return pathlib.Path(output_child)
+        Returns:
+            None
+        """
+        return None
```

Comment:
This works, but you might like to investigate other ways of doing mocking if you are interested, for example in test_utils/test_hdfs_mods.py - or see what copilot comes up with!

Reply:
Also Keilan's suggestion https://github.com/ONSdigital/lms-format/blob/main/tests/read/test_mappers.py

I think though that what you've got is probably fine